### ISSUE TITLE
Add an option for static linking?

### ISSUE THREAD

**Issue by mamhoff** [brummer10/guitarix.vst#14]
Thank you for your work here, really nice! I'm using Bitwig in Fedora, installed as a Flatpak - and Flatpak installations have no access to most system libraries, and loading the plugin fails with 

```
[2024-01-14 19:06:46.579 indexer info] Could not read metadata for Guitarix.vst3 com.bitwig.flt.library.metadata.reader.exception.CouldNotReadMetadataException: could not read metadata: com.bitwig.flt.library.metadata.reader.exception.CouldNotReadMetadataException: could not read metadata: 
Failed to load VST 3 plug-in /home/anselm/.vst3/Guitarix.vst3: libgiomm-2.4.so.1: cannot open shared object file: No such file or directory
```

`libgiomm` is installed, but not available. Something like `make static`? Pretty please?

**Comment by brummer10** [comment]
linking libgiomm statically will. if it even works, result in a very huge binary.
Why don't you use the provided binary from here?
https://github.com/brummer10/guitarix.vst/releases/download/Latest/Guitarix.vst3.zip

**Comment by ssssam** [comment]
I have the same issue. The problem is that the Flatpak container for the Bitwig app does not have libgiomm-2.4 available, so the plugin can't be loaded.

I am not sure static linking is the answer. Fedora doesn't package a static version of libgiomm2.4, so we'd have to (try to) build it from source ourselves, and I have no idea if the link would actually work anyway.

Another solution would be to package Guitarix.vst as a LinuxAudio Flatpak extension, so we could bundle libgiomm2.4.so along with it. There are examples of this already (mostly LV2 plugins), see: <https://github.com/orgs/flathub/repositories?q=linuxaudio&type=all&language=&sort=>. The Bitwig Flatpak already supports the LinuxAudio extension point, see the [manifest](https://github.com/flathub/com.bitwig.BitwigStudio/blob/4723af7e5d9b38294b565b20d6ad64b22cf2be41/com.bitwig.BitwigStudio.yaml#L51C2-L51C37).

**Comment by ssssam** [comment]
Looking at https://github.com/flathub/org.guitarix.Guitarix, there is already a LinuxAudio plugin defined for the LV2 version. That might be a good place to also define a VST extension.

**Comment by brummer10** [comment]
That seems to be doable. But I'm not the one who will do it. Maybe @hfiguiere have a interest in it, as he do the guitarix flatpack as well.

**Comment by hfiguiere** [comment]
I removed giomm from the lv2 plugins on Guitarix, the same should be done in the vst.

**Comment by brummer10** [comment]
guitarix.vst is the full guitarix stack, including preset handling. So removing giomm from the vst version is only possible by  removing giomm from guitarix itself.
guitarix.vst is just a juce wrapper around guitarix engine.

**Comment by hfiguiere** [comment]
This is the problem. You can't link random libraries and expect them to be there. In that specific case (Flatpak) gio is a-okay because it's in the runtime, but giomm is a definite no-no.

**Comment by hfiguiere** [comment]
Otherwise you can link statically and make sure the symbols are not visible.

**Comment by brummer10** [comment]
linking giomm statically is no option.
It may be doable to use the gio C version instead giomm, when that helps.

**Comment by ssssam** [comment]
I actually did a prototype Flatpak extension for Guitarix.vst after writing this. It loaded in Bitwig inside Flatpak, but I saw at least 1 crash and an another issue where audio cutout. I'm not immediately sure how to debug those - I guess if I run Bitwig from the commandline I might see the backtrace at least. But just to temper excitement that there may be more to solve than just packaging / linking issues before we can have Guitarix in Bitwig :-)

**Comment by brummer10** [comment]
There are currently 2 plugs which I need to disable for the vst version, those are the JCM800 and the Fuzz Master.
Are your issues maybe related to those?

**Comment by ssssam** [comment]
Not sure- I didn't specifically open either of those. I did try a couple of presets though. Was late last night so i wasn't being very methodological :-)

**Comment by ssssam** [comment]
Just did some more testing to see if I can reproduce the issues.

In an empty Bitwig project with just "Test tone" going into Guitarix.vst, I tried each effect in turn, and every preset. All worked perfectly, no crashes or dropouts. (Including Fuzz Master :-)

Tried using Guitarix in a real session,audio stopped after using "Distortion - Iron Man" preset - but this does contain Fuzz Master.

Tried a few things after restarting, avoiding Fuzz Master and this time, no issues. So you could be right about the cause here. (Also - Guitarix sounds way better on these guitars than the Bitwig Amp effect, I'm very happy to finally be able to use it here :-)

**Comment by brummer10** [comment]
That's good to hear. 
So I'll going to disable those plugs. Will check if I found a solution for the other issue.

**Comment by hfiguiere** [comment]
now that this is closed, it will be virtually impossible to package this plugin as a flatpak because it's not self contained.
